.github/workflows/benchmarks.yml

```
@@ -0,0 +1,47 @@
+name: "Benchmark workflow"
+on:
+  # Push and update benchmarks when a commit is merged into master.
+  push:
+    branches:
+      - master
+  # Trigger benchmark test on this PR's commit against master.
+  pull_request:
+    branches:
+      - master
+permissions:
+  # Push benchmark performance graph to gh-pages branch.
+  contents: write
+  deployments: write
+jobs:
+  benchmark:
+    name: Performance regression check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - run: go version
+      - name: Build go-algorand
+        run: scripts/travis/build.sh
+      - name: Run benchmark
+        run: go test ./data/transactions/logic -bench 'BenchmarkUintMath' | tee benchmark_output.txt
+      - name: Push benchmark result
+        if: github.event_name == 'push'
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: Go Benchmark
+          tool: 'go'
+          output-file-path: benchmark_output.txt
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          auto-push: true
+      - name: Store benchmark result
```

@algochoi Minor:  Propose changing verbiage to minimize confusion.

As configured, the step _only_ evaluates the benchmark result diff _without_ persisting any state.

I'm open to other step names provided they don't imply persisting state.


```suggestion
      - name: Evaluate benchmark result
```